robolectric/src/main/java/org/robolectric/internal/bytecode/InstrumentationConfiguration.java

```diff
@@ -187,22 +209,13 @@ public boolean shouldAcquire(String name) {
     // See https://github.com/robolectric/robolectric/issues/521
     if (name.equals("android.R$styleable")) return true;
 
-    return !(
-        name.matches(".*\\.R(|\\$[a-z]+)$")
-            || classesToNotAquire.contains(name)
-            || name.startsWith("java.")
-            || name.startsWith("javax.")
-            || name.startsWith("sun.")
-            || name.startsWith("com.sun.")
-            || name.startsWith("org.w3c.")
-            || name.startsWith("org.xml.")
-            || name.startsWith("org.junit")
-            || name.startsWith("org.hamcrest")
-            || name.startsWith("org.specs2")  // allows for android projects with mixed scala\java tests to be
-            || name.startsWith("scala.")      //  run with Maven Surefire (see the RoboSpecs project on github)
-            || name.startsWith("kotlin.")
-            || name.startsWith("com.almworks.sqlite4java") // Fix #958: SQLite native library must be loaded once.
-    );
+    if (packagesToNotAquire.contains(pkgName)) return false;
```

Comment:
I think we should have the code strictly checking packages (which this does). It's unfortunate that substring now creates a new char[] but I don't think that will affect performance much.